src/cmd_carte.rs

```diff
@@ -0,0 +1,125 @@
+use teloxide::{
+    dispatching::dialogue::{GetChatId, InMemStorage},
+    payloads::SendMessageSetters,
+    prelude::Dialogue,
+    requests::Requester,
+    types::{
+        CallbackQuery, ChatId, InlineKeyboardButton, InlineKeyboardMarkup, Message, MessageId, ReplyMarkup
+    },
+    Bot,
+};
+
+use crate::HandlerResult;
+
+#[derive(Default, Clone, Debug)]
+pub enum CardState {
+    #[default]
+    Start,
+    ChooseCardOption,
+    GiveCard {
+        /// ID of the message querying the target of the /carte.
+        /// Used to delete the message after the selection.
+        message_id: MessageId,
+    },
+    // ReturnCard {
+    //     /// ID of the message.
+    //     /// Used to delete the message after the selection.
+    //     message_id: MessageId,
+    // },
+}
+pub type CarteDialogue = Dialogue<CardState, InMemStorage<CardState>>;
+
+
+// when /carte: message with where the card is (either in office, either in CLIC or the name of the holder)
+// if in CLIC, propose to give the card, or to do nothing
+// if a holder have the card, propose to give back the card to the office or do nothing
+
+
+/// Starts the /carte dialogue.
+pub async fn start_card_dialogue(
+    bot: Bot,
+    msg: Message,
+    dialogue: CarteDialogue,
+) -> HandlerResult {
+    log::info!("Starting /carte dialogue");
+
+    // log::debug!("Removing /carte message");
+    // bot.delete_message(msg.chat.id, msg.id).await?;
+
+    // if there is a holder:
+    let cartd_status = "CLIC"; // TODO get DB info from who hold the card
+
+    let is_at_office = cartd_status == "CLIC";
+    let (text, token) = if is_at_office {("Donner la carte", "give_card")} else {("Rendre la carte", "return_card")};
+    
+    let row = vec![
+        InlineKeyboardButton::callback(text, token),
+        InlineKeyboardButton::callback("Cancel", "nothing")
+    ];
```

Comment:
I think it would be good to also have the possibility to "forward" the card, e.g. when a commission holds it, they can give it to another one.